app/models/concerns/provider_twitter.rb

```diff
@@ -3,22 +3,57 @@
 module ProviderTwitter
   extend ActiveSupport::Concern
 
+  class ApiError < StandardError; end
+
+  BASE_URI = "https://api.twitter.com/2/"
+
   def oembed_url(_ = nil)
     "https://publish.twitter.com/oembed?url=#{self.url}"
   end
 
+  def tweet_lookup(tweet_id)
+    params = {
+      "ids": tweet_id,
+      "tweet.fields": "author_id,created_at,text",
+      "expansions": "author_id,attachments.media_keys",
+      "user.fields": "profile_image_url,username,url",
+      "media.fields": "url",
+    }
+
+    get "tweets", params
+  end
+
+  def user_lookup_by_username(username)
+    params = {
+      "usernames": username,
+      "user.fields": "profile_image_url,name,username,description,created_at,url",
+    }
+
+    get "users/by", params
+  end
+
   private
 
-  def handle_twitter_exceptions
+  def get(path, params)
+    uri = URI(URI.join(BASE_URI, path))
+    uri.query = Rack::Utils.build_query(params)
+    
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+
+    headers = {
+      "Authorization": "Bearer #{PenderConfig.get('twitter_bearer_token')}",
+    }
+    
+    request = Net::HTTP::Get.new(uri.request_uri, headers)
+
     begin
-      yield
-    rescue Twitter::Error::TooManyRequests => e
-      raise Pender::Exception::ApiLimitReached.new(e.rate_limit.reset_in)
-    rescue Twitter::Error => error
-      PenderSentry.notify(error, url: url)
-      @parsed_data[:raw][:api] = { error: { message: "#{error.class}: #{error.code} #{error.message}", code: Lapis::ErrorCodes::const_get('INVALID_VALUE') }}
-      Rails.logger.warn level: 'WARN', message: "[Parser] #{error.message}", url: url, code: error.code, error_class: error.class
-      return
+      response = http.request(request)
+      raise ApiError.new("#{response.class}: #{response.code} #{response.message} - #{response.body}") unless response.code.to_i < 400
```

Comment:
**suggestion**: I think we should remove response.body from this error message construction because it should give us a unique error each time in Sentry. It would be helpful to make sure that the response body is sent as an attribute to Sentry, though, for easier debugging!